force-app/main/triggers/classes/HOT_ServiceAppointmentHandler.cls

```diff
@@ -318,8 +319,31 @@ public without sharing class HOT_ServiceAppointmentHandler extends MyTriggers {
                     WHERE Id IN :serviceResourceByServiceAppointmentSchedTimeChanged.values()
                 ]
             );
+
+            Map<Id, ServiceAppointment> oldServiceAppointmentsWithRequestFields = new Map<Id, ServiceAppointment>(
+                [
+                    SELECT
+                        Id,
+                        SchedStartTime,
+                        SchedEndTime,
+                        WorkTypeId,
+                        ServiceTerritoryId,
+                        Status,
+                        HOT_Request__r.ServiceResourceWithAgreement1__c,
+                        HOT_Request__r.ServiceResourceWithAgreement2__c,
+                        HOT_Request__r.ServiceResourceWithAgreement3__c
+                    FROM ServiceAppointment
+                    WHERE Id IN :oldServiceAppointmentsSchedTimeChanged.keySet()
+                ]
+            );
+
+            for (ServiceAppointment sa : oldServiceAppointmentsWithRequestFields.values()) {
+                sa.SchedStartTime = oldServiceAppointmentsSchedTimeChanged.get(sa.Id).SchedStartTime;
+                sa.SchedEndTime = oldServiceAppointmentsSchedTimeChanged.get(sa.Id).SchedEndTime;
+            }
+
```

Comment:
Lurer på om vi skal flytte dette til to metoder, slik at det ikke blir så mange linjer i onAfterUpdate/onBeforeUpdate-metodene.
getServiceAppointmnetsWithAgreementFieldsFromRequest()
setScheduledTimesToOldValues()

noe sånt?